src/Database/Type/DateTimeType.php

```diff
@@ -103,7 +103,7 @@ public function toDatabase($value, Driver $driver)
      */
     public function toPHP($value, Driver $driver)
     {
-        if ($value === null) {
+        if ($value === null || (int)$value === 0) {
```

Comment:
According to some sources `The concept of a year "zero" is a modern myth (but a very popular one). In our calendar, C.E. 1 follows immediately after 1 B.C.E.`
So we dont have to support `0000-...` :trollface: